src/tracks/playabletrack/wavetrack/ui/WaveformCache.cpp

```diff
@@ -90,15 +90,17 @@ bool WaveClipWaveformCache::GetWaveDisplay(
       pWhere = &display.ownWhere;
    }
    else {
-      const double tstep = 1.0 / pixelsPerSecond;
-      const auto rate = clip.GetRate();
-      const double samplesPerPixel = rate * tstep;
+      const auto sampleRate = clip.GetRate();
+      const auto stretchRatio = clip.GetStretchRatio();
+      const double samplesPerPixel =
+         sampleRate / pixelsPerSecond / stretchRatio;
 
       // Make a tolerant comparison of the pps values in this wise:
       // accumulated difference of times over the number of pixels is less than
       // a sample period.
-      const bool ppsMatch = waveCache &&
-         (fabs(tstep - 1.0 / waveCache->pps) * numPixels < (1.0 / rate));
+      const bool ppsMatch =
```

Comment:
The variable name might be updated.  `sppMatch` perhaps.  And update "pps" in the comment.